src/bots/telegram.ts

```diff
@@ -482,35 +487,16 @@ export default class TelegramBot extends BotClient {
     return true;
   }
 
-  // TODO: Properly type the error
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private handleNotificationError(error: any, channel: Channel) {
-    if (error.code === 'ETELEGRAM') {
-      const response = error.response.body;
-      const errorCode = response.error_code;
-      switch (errorCode) {
-        // Chat not found
-        case 400:
-          this.logger.warn(
-            `Failed to send notification to channel ${channel.label}, error code 400.`,
-          );
-          // this.removeData(channel);
-          break;
-        // Bot is not a member of the channel chat or blocked by user
-        case 403:
-          this.logger.warn(
-            `Failed to send notification to channel ${channel.label}, error code 403.`,
-          );
-          // this.removeData(channel);
-          break;
-        default:
-          this.logger.error(
-            `Failed to send notification to channel ${channel.label}, error code ${errorCode}:\n${error}`,
-          );
-      }
-    } else {
-      this.logger.error(`Failed to send message to channel ${channel.label}:\n${error}`);
-    }
+  private handleNotificationError(error: TelegramError, channel: Channel) {
+    const errorCode = error.response.error_code;
+    this.logger.error(
+      `Failed to send notification to channel ${channel.label}, error code ${errorCode}:\n${error}`,
+    );
+    // TODO: conditions
+    // if (conditions) {
+    //   this.logger.warn(`Unable to send notification to channel ${channel.label}, deleting data...`);
+    //   this.removeData(channel);
+    // }
```

Comment:
What about this? If we don't resolve this now we should create an issue

Reply:
This has been commented out for a very long time, I'm not sure if and how it can be addressed and I still think deleting subscribers automatically is bad practice. We still have the automatic disable in place